### PR TITLE
2 week debugging refactoring hooks

### PR DESCRIPTION
**Pull request by OlegBalabukh**

### CODE REVIEW

**Comment by OlegBalabukh** [comment]
Task _"rewrite the app using hooks"_ isn't finished ( I had some issues with SearchForm and WeatherTable components)

File: src/components/App/App.js [hunk 1]
```diff
@@ -1,48 +1,35 @@
...
-  }
 }
 
 App.propTypes = {
-  weather: PropTypes.shape({
-    city: PropTypes.string.isRequired,
-    temp: PropTypes.number.isRequired,
-    isCelsius: PropTypes.bool
-  })
+  stateFromSearchForm: PropTypes.func
```
Comment (odere) [inline src/components/App/App.js]: This props should be in `SearchForm`

File: src/components/SearchForm/SearchForm.js [hunk 2]
```diff
@@ -22,47 +23,44 @@ class SearchForm extends Component {
...
-    let prevIsCelsius = this.state.isCelsius;
+    const prevIsCelsius = this.state.isCelsius;
     this.setState(initialState);
     this.setState({isCelsius: prevIsCelsius});
   }
 
   getState = (radioButtonsState) => {
-    radioButtonsState.value === "C" ? 
-    this.setState({isCelsius: true}) : 
+    radioButtonsState === "C" ?
```
Comment (odere) [inline src/components/SearchForm/SearchForm.js]: use constant

File: src/components/WeatherTable/WeatherTable.css [hunk 3]
```diff
@@ -1,14 +1,17 @@
 .rootTable {
   width: 400px;
   margin: 0 auto;
-  margin-top: 20px;  
+  margin-top: 20px;
```
Comment (odere) [inline src/components/WeatherTable/WeatherTable.css]: Please use only margin

Comment (OlegBalabukh) [reply src/components/WeatherTable/WeatherTable.css]: fixed

File: src/components/SearchForm/SearchForm.js (hunk 2 above)
Comment (OlegBalabukh) [reply src/components/SearchForm/SearchForm.js]: fixed

File: src/components/App/App.js (hunk 1 above)
Comment (OlegBalabukh) [reply src/components/App/App.js]: fixed

**Pull request merged**